File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans=new int[2];
        Map<Integer,Integer> map=new HashMap<Integer,Integer>();
        for(int i=0;i<nums.length;i++){
            if(map.containsKey(target-nums[i])){
                ans[0]=map.get(target-nums[i]);
                ans[1]=i;
                return ans;
            }else{
                map.put(nums[i],i);
            }
        }
        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let mp=new Map();
    let ans = new Array(2).fill(0);

    for(let i=0;i<nums.length;i++){
        if(mp.has(target-nums[i])){
            ans[0]=mp.get(target-nums[i]);
            ans[1]=i;
            return ans;
        }
        mp.set(nums[i],i);
    }
    return ans;
};